### PR TITLE
Adapt to recent version of PHPUnit (10)

### PR DESCRIPTION
**Pull request by DavidPrevot**
Those changes allow the testsuite to also pass on PHPUnit 10.

Regards,

David

### CODE REVIEW

**Comment by greg0ire** [comment]
Hi David! It looks like you forgot to bump the version constraint here: https://github.com/doctrine/instantiator/blob/d6eef505a6c46e963e54bf73bb9de43cdea70821/composer.json#L28

In case you're wondering, this is why the checks are red.

Please make the change, amend your commit and force push.

**Comment by DavidPrevot** [comment]
Thanks for the hint (I thought I actually run the testsuite with PHPUnit 9 after the change, but obviously forgot).

**Comment by greg0ire** [comment]
You're welcome. I believe these lines need to be changed as well: https://github.com/doctrine/instantiator/blob/2.0.x/phpunit.xml.dist#L22-L26

They use old vocabulary (`filter`/ `whitelist` vs `coverage` / `include`)

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,27 +1,15 @@
...
-
-    <testsuite name="Doctrine\Instantiator tests">
-        <directory>./tests/DoctrineTest/InstantiatorTest</directory>
-    </testsuite>
-    <filter>
-        <whitelist addUncoveredFilesFromWhitelist="true">
-            <directory suffix=".php">./src</directory>
-        </whitelist>
-    </filter>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.1/phpunit.xsd" bootstrap="./vendor/autoload.php" colors="true" stopOnFailure="false" processIsolation="false" backupGlobals="false" cacheDirectory=".phpunit.cache">
```
Comment (greg0ire) [inline phpunit.xml.dist]: Can you please use linebreaks here? Like in the previous version?

File: phpunit.xml.dist [hunk 2]
```diff
@@ -1,27 +1,23 @@
 <?xml version="1.0"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+    xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.1/phpunit.xsd"
```
Comment (greg0ire) [inline phpunit.xml.dist]: Please revert this change, IMO it's more guaranteed to be the right version of the xsd file here.

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @DavidPrevot !
